lib/core/src/swapper/boltz/mod.rs

```diff
@@ -421,24 +499,27 @@ impl Swapper for BoltzSwapper {
         Ok(tx_id)
     }
 
-    fn create_status_stream(&self) -> Box<dyn SwapperStatusStream> {
-        Box::new(BoltzStatusStream::new(&self.boltz_url))
+    fn create_status_stream(&self) -> Result<Box<dyn SwapperStatusStream>> {
+        Ok(Box::new(BoltzStatusStream::new(&self.get_client()?.url)))
```

Comment:
This will block on creating the client which miss the goal of not blocking the initialization.
Perhaps we can pass the url later when the status stream connects?